model_analyzer/config/generate/perf_analyzer_config_generator.py

```diff
@@ -199,55 +246,148 @@ def _create_concurrency_list(self) -> List[int]:
             )
 
+    def _create_prompt_length_list(self) -> List[int]:
+        if not self._cli_config.is_llm_model():
+            return []
+
+        if self._model_parameters["prompt_length"]:
+            return sorted(self._model_parameters["prompt_length"])
+        elif self._cli_config.run_config_search_disable:
+            return [1]
+        else:
+            return utils.generate_doubled_list(
+                self._cli_config.run_config_search_min_prompt_length,
+                self._cli_config.run_config_search_max_prompt_length,
+            )
+
+    def _create_max_token_count_list(self) -> List[int]:
+        if not self._cli_config.is_llm_model():
+            return []
+
+        if self._model_parameters["max_token_count"]:
+            return sorted(self._model_parameters["max_token_count"])
+        elif self._cli_config.run_config_search_disable:
+            return [1]
+        else:
+            return utils.generate_doubled_list(
+                self._cli_config.run_config_search_min_token_count,
+                self._cli_config.run_config_search_max_token_count,
+            )
+
     def _generate_perf_configs(self) -> None:
-        perf_config_non_parameter_values = (
-            self._create_non_parameter_perf_config_values()
+        all_non_parameter_combinations = utils.generate_parameter_combinations(
+            self._perf_config_non_parameter_values
         )
-
-        for params in utils.generate_parameter_combinations(
-            perf_config_non_parameter_values
-        ):
-            configs_with_concurrency = []
+        for unmodified_non_parameter_combination in all_non_parameter_combinations:
+            all_perf_configs_for_a_given_parameter = []
             for parameter in self._parameters:
-                new_perf_config = PerfAnalyzerConfig()
-
-                new_perf_config.update_config_from_profile_config(
-                    self._model_name, self._cli_config
+                new_perf_config = self._create_new_perf_config(
+                    parameter, unmodified_non_parameter_combination
                 )
+                all_perf_configs_for_a_given_parameter.append(new_perf_config)
 
-                new_perf_config.update_config(params)
+            self._configs.append(all_perf_configs_for_a_given_parameter)
 
-                if self._cli_config.is_request_rate_specified(self._model_parameters):
-                    new_perf_config.update_config({"request-rate-range": parameter})
-                else:
-                    new_perf_config.update_config({"concurrency-range": parameter})
+    def _create_new_perf_config(
+        self, parameter: int, unmodified_non_parameter_combination: Dict
+    ) -> PerfAnalyzerConfig:
+        perf_config = self._create_base_perf_config()
+
+        (
+            prompt_length,
+            modified_non_parameter_combination,
+        ) = self._extract_prompt_length(unmodified_non_parameter_combination)
```

Comment:
I previously did that, but it's a codeQL violation since I mutating a dictionary that's passed by reference. The only way to get rid of the violation was to make a copy of the dict (even using deepcopy caused a violation!) and pass it back as a different variable.

Reply:
That's strange! Thanks for clarifying 👍